Update Sunspot's OneAPI compiler to 2023.05.15.007

amametjanov opened the pull request:
Update Sunspot's OneAPI compiler to 2023.05.15.007 . Also,
- add netcdf libs to LD_LIBRARY_PATH after recent CMake updates
- rearrange modules by loading defaults after module purge
- clean up cmake

Fixes E3SM-Project/E3SM#5707

[BFB]

amametjanov commented:
This version can build and run `SMS_P12x2.ne4_oQU240.WCYCL1850NS.sunspot_oneapi-ifx.allactive-mach_mods`, which was failing previously with an ICE error as in E3SM-Project/E3SM#5522 (also relevant is E3SM-Project/E3SM#5629)

abagusetty commented on cime_config/machines/config_machines.xml:
```diff
@@ -2937,7 +2937,7 @@
         <modules compiler="!gnu">
-          <command name="load">oneapi/eng-compiler/2022.12.30.003</command>
+          <command name="load">oneapi/eng-compiler/2023.05.15.007</command>
           <command name="load">mpich/52.2/icc-all-pmix-gpu</command>
```
@amametjanov Can you also consider removing/commenting the mpich version since a new, compatible version (`mpich/52.2-256/icc-all-pmix-gpu`) of MPICH gets loaded with 2023.05.15.007

rljacob commented:
status:  waiting for CIME update for some tests.

rljacob commented:
status:  Az will update an merge.

amametjanov commented:
Updated the branch and rebased onto latest master.

OneAPI module `2023.05.15.007` fixes previous threading bugs, but breaks debug-mode tests (link errors with new MSAN memory sanitizer libs due to presence of `-check uninit` flags). Fix is expected in future compiler updates.

Current `e3sm_integration` status -- 94 pass, 25 known fails: https://my.cdash.org/viewTest.php?buildid=2421641 .
